crates/astria-conductor/src/sequencer/sync.rs

```diff
@@ -0,0 +1,102 @@
+use astria_sequencer_types::SequencerBlockData;
+use color_eyre::eyre::{
+    self,
+    WrapErr as _,
+};
+use deadpool::managed::{
+    Pool,
+    PoolError,
+};
+use futures::stream::FuturesOrdered;
+use tokio::select;
+use tracing::{
+    error,
+    info,
+    instrument,
+    warn,
+};
+
+use crate::client_provider::{
+    self,
+    ClientProvider,
+};
+
+#[derive(Debug, thiserror::Error)]
+enum Error {
+    #[error("failed requesting a client from the pool")]
+    Pool(#[from] PoolError<client_provider::Error>),
+    #[error("getting a block from sequencer failed")]
+    Request(#[from] sequencer_client::extension_trait::Error),
+}
+
+#[instrument(skip(client_pool))]
+pub(super) async fn run(
+    start: u32,
+    end: u32,
+    client_pool: Pool<ClientProvider>,
+    executor: crate::executor::Sender,
+) -> eyre::Result<()> {
+    use futures::{
+        FutureExt as _,
+        StreamExt as _,
+    };
+
+    let mut height_stream = futures::stream::iter(start..end);
```

Comment:
check if `start > end` and return if so?

Reply:
So this makes use of a bunch of things:
1. assume `start > end`
2. then the iterator impl for `start..end` will always return `None` because it's empty and because it's a `FusedIterator`.
3. this means that `futures::stream::iter` will always resolve to `Ready(None)` immediately
4. this means that the `select!` statement wil poll the stream, immediately get a result, try apply it's result to the `Some(_)` pattern, fail, and hence disable the branch.
5. because the `FuturesOrdered` stream also response with `Ready(None)` if it's empty, its select arm will also be disabled.
6. so because both arms are disabled, we immediately go to `else`